pkgs/swiftgen/swift2objc/lib/src/parser/parsers/declaration_parser.dart

```diff
@@ -0,0 +1,35 @@
+import '../../ast/_core/interfaces/declaration.dart';
+import '../_core/parsed_symbol.dart';
+import '../_core/type_defs.dart';
+import 'declaration_parsers/class_parser.dart';
+import 'declaration_parsers/method_parser.dart';
+
+class DeclarationParser {
+  final ParsedSymbolsMap parsedSymbolsMap;
+  final String parsedSymbolId;
+
+  DeclarationParser(this.parsedSymbolId, this.parsedSymbolsMap);
+
+  ParsedSymbol get parsedSymbol => parsedSymbolsMap[parsedSymbolId]!;
+
+  Declaration parse() {
+    if (parsedSymbol.declaration != null) {
+      return parsedSymbol.declaration!;
+    }
+
+    final symbolJson = parsedSymbol.json;
+
+    final String symbolType = symbolJson["kind"]["identifier"];
+
+    final parser = switch (symbolType) {
+      "swift.class" => ClassParser(parsedSymbolId, parsedSymbolsMap),
+      "swift.method" => MethodParser(parsedSymbolId, parsedSymbolsMap),
+      _ => throw UnsupportedError(
```

Comment:
Not sure if you're catching this further up the stack, but we'll want this to be a non-fatal error, since it's gonna be a while before we support everything.

Reply:
Added a catch for `UnsupportedError` in `AstParser` and a logged a message when it occurs

Reply:
Ok. I think technically this should be a `UnimplementedError`. `UnimplementedError` means "we haven't built this yet", but `UnsupportedError` means "we're never going to support this".

Reply:
done